src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,230 @@
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Manifest;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.Oracle
+{
+    public sealed class OraclePolicyContract : NativeContract
+    {
+        public override string ServiceName => "Neo.Native.Oracle.Policy";
+
+        public override int Id => -4;
+
+        private const byte Prefix_Validator = 24;
+        private const byte Prefix_TimeOutMilliSeconds = 25;
+        private const byte Prefix_PerRequestFee = 26;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public OraclePolicyContract()
+        {
+            Manifest.Features = ContractFeatures.HasStorage;
+        }
+
+        /// <summary>
+        /// Initialization.Set default parameter value.
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <returns>Returns true if the execution is successful, otherwise returns false</returns>
+        internal override bool Initialize(ApplicationEngine engine)
+        {
+            if (!base.Initialize(engine)) return false;
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_TimeOutMilliSeconds), new StorageItem
+            {
+                Value = BitConverter.GetBytes(5000)
+            });
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_PerRequestFee), new StorageItem
+            {
+                Value = BitConverter.GetBytes(1000)
+            });
+            return true;
+        }
+
+        /// <summary>
+        /// Oracle validator can delegate third party to operate Oracle nodes
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Returns true if the execution is successful, otherwise returns false</returns>
+        [ContractMethod(0_01000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.ByteArray, ContractParameterType.ByteArray }, ParameterNames = new[] { "consignorPubKey", "consigneePubKey" })]
+        private StackItem DelegateOracleValidator(ApplicationEngine engine, Array args)
+        {
+            ECPoint consignorPubKey = args[0].GetSpan().AsSerializable<ECPoint>();
+            ECPoint consigneePubKey = args[1].GetSpan().AsSerializable<ECPoint>();
+            UInt160 account = Contract.CreateSignatureRedeemScript(consignorPubKey).ToScriptHash();
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
+            StoreView snapshot = engine.Snapshot;
+            StorageKey key = CreateStorageKey(Prefix_Validator, consignorPubKey);
```

Comment:
It may be better to use the validator index for the prefix than the public key. In this way, when changing a consensus node we prevent to left the delegation record taking up space.

Reply:
This may not be very safe.Because the new consensus node needs to manually set the delegation, if the new joiner forgets to reset the delegation, there will be risks.If the setting is automatically changed, it will increase the consensus operation.

Reply:
We can store in the value who was the consignor, in order to skip if it's not the same, otherwise we can leave for the eternity some entries in the storage.

Reply:
> This may not be very safe.Because the new consensus node needs to manually set the delegation

I didn't see a problem with this because it's the new validator's responsibility to update the record, but you are right it is always better to avoid mistakes.

What other way can you think of to not leave "leftovers" in storage when there is a change of consensus?

Reply:
> it's the new validator's responsibility to update the record

But it's not very convenient, new oracle nodes need to be manually updated every time. 
I agree that maybe we should delete these "leftovers" records.

Reply:
I can do a check in delegate function and delete the leftovers.